plugins/Utils/windowstatestorage.cpp

```diff
@@ -16,167 +17,319 @@
 
 #include "windowstatestorage.h"
 
-#include <QtConcurrent>
 #include <QDebug>
+#include <QDir>
+#include <QMetaObject>
+#include <QObject>
 #include <QSqlQuery>
 #include <QSqlError>
 #include <QSqlResult>
+#include <QStandardPaths>
 #include <QRect>
 #include <unity/shell/application/ApplicationInfoInterface.h>
 
-QMutex WindowStateStorage::s_mutex;
+AsyncQuery::AsyncQuery(const QString& dbName):
+    m_dbName(dbName)
+{
+}
 
-inline QString sanitiseString(QString string) {
-    return string.remove(QLatin1Char('\"'))
-                 .remove(QLatin1Char('\''))
-                 .remove(QLatin1Char('\\'));
+AsyncQuery::~AsyncQuery()
+{
+    QSqlDatabase::removeDatabase(m_connectionName);
```

Comment:
Is this always safe to run? even if there is no database to remove?

Reply:
Yes.